scenario_player/runner.py

```diff
@@ -153,289 +335,326 @@ def local_seed(self) -> str:
 
-            setup = nursery.spawn_under_watch(
-                self._setup, mint_gas, node_count, node_addresses, fund_tx, node_starter
+    def setup_raiden_nodes_ether_balances(
+        self, pool: Pool, node_addresses: Set[ChecksumAddress]
+    ) -> Set[Greenlet]:
+        """ Makes sure every Raiden node has at least `NODE_ACCOUNT_BALANCE_MIN`. """
+
+        greenlets: Set[Greenlet] = set()
+        for address in node_addresses:
+            g = pool.spawn(
+                eth_maybe_transfer,
+                self.client,
+                to_canonical_address(address),
+                NODE_ACCOUNT_BALANCE_MIN,
             )
-            setup.name = "setup (root task)"
-            greenlets = {setup}
+            greenlets.add(g)
 
-            gevent.joinall(greenlets, raise_error=True, count=1)
+        return greenlets
 
-    def _initialize_scenario_token(
+    def setup_mint_user_deposit_tokens_for_distribution(
         self,
+        pool: Pool,
+        userdeposit_proxy: UserDeposit,
+        token_proxy: CustomToken,
         node_addresses: Set[ChecksumAddress],
-        udc_ctr: Optional[Contract],
-        should_deposit_ud_token: bool,
-        gas_limit: int,
-    ) -> Set[TransactionHash]:
-        """Create or reuse an existing token, and mint the token for every
-        `node_addresses`.
+    ) -> Set[Greenlet]:
+        """ Ensures the scenario player account has enough tokens and allowance
+        to fund the Raiden nodes.
         """
-        self.token.init()
-        mint_tx = set()
-        for address in node_addresses:
-            tx = self.token.mint(address)
-            if tx:
-                mint_tx.add(tx)
+        settings = self.definition.settings
+        udc_settings = settings.services.udc
+        balance_per_node = settings.services.udc.token.balance_per_node
+
+        msg = "udc is not enabled, this function should not be called"
+        assert is_udc_enabled(udc_settings), msg
+
+        node_count = len(node_addresses)
+        required_allowance = balance_per_node * node_count
+
+        allowance_greenlet = pool.spawn(
+            userdeposit_maybe_increase_allowance,
+            token_proxy=token_proxy,
+            userdeposit_proxy=userdeposit_proxy,
+            orchestrator_address=self.client.address,
+            minimum_allowance=required_allowance,
+            maximum_allowance=UINT256_MAX,
+        )
 
-            if not should_deposit_ud_token:
-                continue
-            assert self.udc
-            deposit_tx = self.udc.deposit(address)
-            if deposit_tx:
-                mint_tx.add(deposit_tx)
+        mint_greenlet = pool.spawn(
+            token_maybe_mint,
+            token_proxy,
+            to_checksum_address(self.client.address),
+            minimum_balance=required_allowance,
+            maximum_balance=ORCHESTRATION_MAXIMUM_BALANCE,
+        )
 
-        return mint_tx
+        return {allowance_greenlet, mint_greenlet}
```

Comment:
Same question as above

Reply:
This one is necessary, this thread mints tokens for the orchestration account, which is then transferred by `setup_raiden_nodes_with_sufficient_user_deposit_balances`. These greenlets are used for synchronization since the token transfer can only be performed after the mint transaction is